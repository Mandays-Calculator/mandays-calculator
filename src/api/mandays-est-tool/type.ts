export interface TasksResponse {
  id: string;
  name: string;
  description: string;
  status: string;
  functionality: Functionality;
  tags: string;
  comments: Comments[];
  sprint: string;
  completionDate: string;
  createdDate: string;
}

interface Functionality {
  id: string;
  name: string;
}

interface Comments {
  id: string;
  description: string;
}
