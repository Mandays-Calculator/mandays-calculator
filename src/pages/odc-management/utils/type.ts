export type DataType = {
  odcName: string;
  location: string;
  abbreviation: string;
  noHolidays: number;
};

export type HolidayType = {
  holidayDate: string;
  holidayName: string;
};