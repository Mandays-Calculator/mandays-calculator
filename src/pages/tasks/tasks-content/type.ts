export interface Task {
    taskID: string;
    taskTitle: string;
    desc: string;
    date: string;
    sprint: string;
    complexity: string;
    status: string;
    type: string;
    functionality: string;
    tags: string[];
    comments: {
        name: string;
        comment: string;
    }[];
}