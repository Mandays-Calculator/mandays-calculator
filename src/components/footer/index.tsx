export { default as Footer } from "./Footer";
export { StyledFooterContainer, StyledFooterText } from "./styles";
