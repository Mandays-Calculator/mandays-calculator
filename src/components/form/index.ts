export * from "./checkbox";
export * from "./checkbox-group";
export * from "./text-field";
export * from "./date-picker";
