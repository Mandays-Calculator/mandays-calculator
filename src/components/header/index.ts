export {
  StyledAppBar,
  StyledToolBarContainer,
  StyledSearchSelect,
  popperStyle,
  StyledProjectTitle,
} from "./styles";
