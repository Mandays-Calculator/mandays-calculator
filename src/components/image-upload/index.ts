export { default as ImageUpload } from "./ImageUpload";
export * from "./styles";
