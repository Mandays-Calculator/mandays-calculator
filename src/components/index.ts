export * from "./form/text-field";
export * from "./form/Form";
export * from "./title";
export * from "./svc-icons";
export * from "./form/select";
export * from "./table";
export * from "./not-found";
export * from "./something-wrong";
