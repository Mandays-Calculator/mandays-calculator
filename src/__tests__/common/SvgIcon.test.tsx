import type { SvgIconsType } from "~/components/svc-icons/types";

import { render } from "@testing-library/react";
import { SvgIcon } from "~/components";

const variants: SvgIconsType[] = [
  "delete",
  "edit",
  "check",
  "add",
  "cross",
  "eyball_1",
  "odc_management",
  "people",
  "person",
  "project_management",
  "settings",
  "user_management",
];
describe("Svg Icon Component", () => {
  it("renders the svg icon", () => {
    variants.forEach((variant) => {
      const { getByTestId, unmount, asFragment } = render(<SvgIcon name={variant} />);

      const icon = getByTestId("svg-component");
      expect(icon).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
      unmount();
    });
  });
});
