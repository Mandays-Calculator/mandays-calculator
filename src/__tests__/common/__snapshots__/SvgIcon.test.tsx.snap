// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg Icon Component renders the svg icon 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-delete"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 2`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-edit"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 3`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-check"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 4`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-add"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 5`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-cross"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 6`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-eyball_1"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 7`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-odc_management"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 8`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-people"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 9`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-person"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 10`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-project_management"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 11`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-settings"
    />
  </svg>
</DocumentFragment>
`;

exports[`Svg Icon Component renders the svg icon 12`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
    data-testid="svg-component"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="#icon-user_management"
    />
  </svg>
</DocumentFragment>
`;
